nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/AbstractJsonEnricher.java

```diff
@@ -56,8 +64,20 @@ public boolean accept(Class<?> clazz, Type genericType, MediaType mediatype) {
 
     @Override
     public void write(Enriched<EntityType> enrichable, JsonGenerator jg) {
-        try {
-            write(jg, enrichable.getEntity());
+        try (Writer writer = new StringWriter()) {
+            try (TokenBuffer tb = new TokenBuffer(MAPPER, false)) {
+                // Write to a temporary output
+                tb.writeStartObject();
+                write(tb, enrichable.getEntity());
+                tb.flush();
+                // Add the content to the real output
+                try (JsonParser parser = tb.asParser()) {
+                    parser.nextToken(); // ignoring START_OBJECT
+                    while (parser.nextToken() == JsonToken.FIELD_NAME) {
+                        jg.copyCurrentStructure(parser);
+                    }
```

Comment:
I think it's worth checking that we're really at the end here after the loop, and if not at least log an error with the enricher name (code untested):

    if (parser.currentToken() != null) {
        log.error("Enricher {} returned invalid output {}", name, tb.toString());
    }